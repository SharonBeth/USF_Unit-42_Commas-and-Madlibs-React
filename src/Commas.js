import React, { useState } from "react";
import NumbersForm from "./NumbersForm";
import "./Commas.css"

function Commas() {

    const [entry, setEntry] = useState(0)

    const create = newEntry => {
        newEntry = Number.parseFloat(newEntry)
        let nextStep = newEntry.toLocaleString();
        setEntry(nextStep);
    };


    return (
        <div className="Commas">
            <NumbersForm addCommas={create} />
            <p>{entry}</p>
        </div>
    )
}

export default Commas;