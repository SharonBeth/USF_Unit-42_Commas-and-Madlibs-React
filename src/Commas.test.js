import React from "react";
import Commas from "./Commas";
import { render, fireEvent } from "@testing-library/react";



// describe("#Commas", () => {
// test("it is a function", () => {
// expect(typeof Commas).toBe("function");
// });
// });

it("renders without crashing-Smoke Test", function () {
  render(<Commas />)
});

it("passes a snapshot", function () {
  const { asFragment } = render(<Commas />)
  expect(asFragment()).toMatchSnapshot();
});

// it("should add commas", function () {
  // const { queryByText, getByLabelText } = render(<Commas />)
  // const input = getByLabelText("Change the Number to a string and add commas:");
  // const btn = queryByText("Submit Numbers")
  // expect(queryByText(",")).not.toBeInTheDocument();
// 
  // fireEvent.change(input, { target: { value: '1000' } });
  // fireEvent.click(btn, { target: { value: '1000' } });
  // const { asFragment } = render(<Commas />)
  // expect(asFragment()).toMatchSnapshot();
  // expect(queryByText(",")).toBeInTheDocument();
// })