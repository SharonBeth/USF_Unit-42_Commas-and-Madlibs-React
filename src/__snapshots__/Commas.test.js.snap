// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes a snapshot 1`] = `
<DocumentFragment>
  <div
    class="Commas"
  >
    <div>
      <form>
        <label
          for="change"
        >
          Change the Number to a string and add commas:
        </label>
        <input
          id="change"
          name="testing"
          placeholder="number"
          type="number"
          value="0"
        />
        <button>
          Submit Numbers
        </button>
      </form>
    </div>
    <p>
      0
    </p>
  </div>
</DocumentFragment>
`;

exports[`should add commas 1`] = `
<DocumentFragment>
  <div
    class="Commas"
  >
    <div>
      <form>
        <label
          for="change"
        >
          Change the Number to a string and add commas:
        </label>
        <input
          id="change"
          name="testing"
          placeholder="number"
          type="number"
          value="0"
        />
        <button>
          Submit Numbers
        </button>
      </form>
    </div>
    <p>
      0
    </p>
  </div>
</DocumentFragment>
`;
